src/averagePerceptronTagger.jl

```diff
@@ -0,0 +1,326 @@
+using DataStructures
+using Random
+using BSON
+
+"""AVERAGE PERCEPTRON MODEL"""
+
+"""
+This module contains the Average Perceptron model
+which was original implemented by Matthew Honnibal.
+
+The model learns by basic perceptron algorithm
+but after all iterations weights are being averaged
+"""
+mutable struct AveragePerceptron
+    classes :: Set
+    weights :: Dict
+    _totals :: DefaultDict
+    _tstamps :: DefaultDict
+    i :: Int64
+    START :: Array
+
+    predict :: Function
+    update :: Function
+    average_weights :: Function
+
+    function AveragePerceptron()
+        self = new()
+
+        self.classes = Set()
+        self.weights = Dict()
+        self._totals = DefaultDict(0)
+        self._tstamps = DefaultDict(0)
+        self.i = 1
+        self.START = ["-START-", "-START2-"]
+
+        self.predict = function (features)

```

Comment:
So while julia does have some object oriented features, generally code in Julia should be written in a more functional style. The biggest difference is, objects do not contain methods inside them. So `predict` should not be a method inside the object. Instead, write this as a function definition at the top level (not inside the type), and let the function take an instance of the `AvergagePerceptron` as the first parameter. Do this for all the functions below. 

```
function predict(ap::AveragePerceptron, features)
    <body of predict function>
end
```

Reply:
This happened due to working for long time in object-oriented programming languages. I will these changes.